File: Iteration2/code/CSE3063GRP12/src/core/enums/AdvisorMenu.java
package core.enums;

import features.advisor.course_approval.CourseApprovalController;
import features.advisor.student_list.AdvisorStudentListController;
import features.login.LoginController;
import features.notification.NotificationController;

public enum AdvisorMenu implements Menu {
    CourseApproval, StudentList, Notification, Logout;

    public String getItemMessage() {
        switch (this) {
            case CourseApproval:
                return ("Course Approval");
            case StudentList:
                return ("Student List");
            case Notification:
                return ("Notification");
            case Logout:
                return ("Logout");
        }
        return this.name();
    }
    public boolean isNotification(){
        return this == Notification;
    }
    public void navigate() {
        switch (this) {
            case CourseApproval:
                new CourseApprovalController();
                break;
            case StudentList:
                new AdvisorStudentListController();
                break;
            case Notification:
                new NotificationController();
                break;
            case Logout:
                new LoginController();
                break;
        }
        
    }
    
}

File: Iteration3/code/CSE3063GRP12/src/core/enums/AdvisorMenu.py
from core.enums.Menu import Menu
from features.notification.NotificationController import NotificationController
from features.advisor.course_approval.CourseApprovalController import CourseApprovalController
from features.advisor.student_list.AdvisorStudentListController import AdvisorStudentListController

class AdvisorMenu(Menu):
    COURSE_APPROVAL = "Course Approval"
    STUDENT_LIST = "Student List"
    NOTIFICATION = "Notification"
    LOGOUT = "Logout"

    def get_item_message(self):
        return self.value
    
    def navigate(self):
        if self == AdvisorMenu.COURSE_APPROVAL:
            CourseApprovalController()
        elif self == AdvisorMenu.STUDENT_LIST:
            AdvisorStudentListController()
        elif self == AdvisorMenu.NOTIFICATION:
            NotificationController()
        elif self == AdvisorMenu.LOGOUT:
            from features.login.LoginController import LoginController
            LoginController()  

    def is_notification(self):
        return self == AdvisorMenu.NOTIFICATION